File: src/main/java/com/gitlab/daring/image/sandbox/MainPanel.java
package com.gitlab.daring.image.sandbox;

import com.gitlab.daring.image.command.CommandScriptPanel;
import com.gitlab.daring.image.command.parameter.FileParam;
import com.gitlab.daring.image.command.parameter.StringParam;

class MainPanel extends CommandScriptPanel {

    final ImageSandbox sb;
    final StringParam filesParam;

    MainPanel(ImageSandbox sb) {
        this.sb = sb;
        filesParam = createFilesParam();
        setScript(sb.config.getString("script"));
    }

    StringParam createFilesParam() {
        FileParam p = new FileParam(":Файлы:0").bind(sb.config, "files");
        addStaticParams(p);
        return p;
    }

}

File: src/main/kotlin/com/gitlab/daring/image/sandbox/MainPanel.kt
package com.gitlab.daring.image.sandbox

import com.gitlab.daring.image.command.CommandScriptPanel
import com.gitlab.daring.image.command.parameter.FileParam
import com.gitlab.daring.image.command.parameter.StringParam

internal class MainPanel(val sb: ImageSandbox) : CommandScriptPanel() {

    val filesParam = createFilesParam()

    init {
        setScript(sb.config.getString("script"))
    }

    fun createFilesParam(): StringParam {
        val p = FileParam(":Файлы:0").bind<FileParam>(sb.config, "files")
        addStaticParams(p)
        return p
    }

}